### PR TITLE
Notebook for SM Training Compiler TF on Multi Node Multi GPU

### PR DESCRIPTION
**Pull request by Lokiiiiii**
*Description of changes:*
This Notebook demonstrates the capability of SM Training Compiler integrated HuggingFace TF containers in a Multi Node Multi GPU setting for Distributed Training.

*Testing done:*
Notebook runs end to end on a SM Notebook Instance. Attached saved PDF of last run.
[Last Successful Run](https://github.com/Lokiiiiii/amazon-sagemaker-examples-staging/files/7592391/language-modeling-multi-gpu-multi-node.pdf)

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [x] I have tested my notebook(s) and ensured it runs end-to-end
- [x] I have linted my notebook(s) and code using `tox -e black-format,black-nb-format`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sagemaker-training-compiler/huggingface/tensorflow_multiple_gpu_multiple_node/language-modeling.ipynb [hunk 1]
```diff
@@ -0,0 +1,465 @@
+{
```
Comment (mchoi8739) [inline sagemaker-training-compiler/huggingface/tensorflow_multiple_gpu_multiple_node/language-modeling.ipynb:1]: <p>There's no native training estimator..</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3053/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (mchoi8739) [inline sagemaker-training-compiler/huggingface/tensorflow_multiple_gpu_multiple_node/language-modeling.ipynb:1]: <p>Add another variable for n_gpu.</p><p> </p><p>Might be better to use <code>ml.p3.16xlarge</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3053/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (jkroll-aws) [reply sagemaker-training-compiler/huggingface/tensorflow_multiple_gpu_multiple_node/language-modeling.ipynb:1]: <p>Also your comment (  <code># ml.p3.8xlarge is easily available. However, ml.p3.16xlarge provides better performance.</code>  ) does not fit with the use of ml.p3dn.24xlarge. I agree it's better to use ml.p3.16xlarge, and keep comment as-is. If you use the 24xlarge, revise the comment.</p>

Comment (jkroll-aws) [inline sagemaker-training-compiler/huggingface/tensorflow_multiple_gpu_multiple_node/language-modeling.ipynb:1]: <p>Add more detailed computation and comparison between the two billing times? It seems like you're only looking at optimized training time here, not native training time. Relates back to Miyoung's comment that this example does not use a native estimator.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3053/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

File: sagemaker-training-compiler/huggingface/tensorflow_multiple_gpu_multiple_node/language-modeling.ipynb [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Compile and Train the GPT2 Model using the Transformers Trainer API with the SST2 Dataset for Multi-Node Multi-GPU Training\n"
```
Comment (pinaraws) [inline sagemaker-training-compiler/huggingface/tensorflow_multiple_gpu_multiple_node/language-modeling.ipynb:7]: Model is roberta-base, not gpt-2

**Comment by jsheng-jian** [comment]
latest run result
[language-modeling.pdf](https://github.com/aws/amazon-sagemaker-examples/files/7663063/language-modeling.pdf)

**Comment by jsheng-jian** [comment]
[language-modeling.pdf](https://github.com/aws/amazon-sagemaker-examples/files/7678047/language-modeling.pdf)

**Comment by jkroll-aws** [comment]
Waiting on this one for newer TF version.

File: sagemaker-training-compiler/huggingface/tensorflow_multiple_gpu_multiple_node/scripts/train_with_keras.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+import sys
+import json
+
+import tensorflow as tf
+from datasets import load_dataset
+from transformers import AutoTokenizer, TFAutoModelForSequenceClassification
+
+from tensorflow.keras import mixed_precision
+
+# Parse sagemaker tensorflow distributed configuration
```
Comment (Lokiiiiii) [inline sagemaker-training-compiler/huggingface/tensorflow_multiple_gpu_multiple_node/scripts/train_with_keras.py:13]: SageMaker Training ToolKit already seems to be doing this: https://github.com/aws/sagemaker-tensorflow-training-toolkit/blob/master/src/sagemaker_tensorflow_container/training.py#L37. 

Please provide differentiation and justification.

**Comment by zhaoqizqwang** [comment]
Closing this PR since it was created a long time ago and didn't pass all the checks. Feel free to open a new pull request instead.

**Pull request closed without merging**